web/modules/custom/custom_page/tests/src/Kernel/CustomPageOgMenuLinkUpdaterTest.php

```diff
@@ -0,0 +1,239 @@
+    $child_custom_page->set('og_audience', $collection_ids[1])->save();
+
+    // Check that the menu link has been removed from the 2nd collection.
+    $this->assertNotMenuLink($ogmenu_instance_ids[2], $child_custom_page->id());
+    // Check that a menu link has been created in the 1st collection.
+    $child_link = $this->assertMenuLink($ogmenu_instance_ids[1], $child_custom_page->id());
+    // Restore the parent relation by recreating the menu link relationship.
+    $child_link->set('parent', "menu_link_content:{$link->uuid()}")->save();
+
+    // Move the parent custom page to the 2nd collection.
+    $custom_page->set('og_audience', $collection_ids[2])->save();
+
+    // Check that the menu link has been removed from the 1st collection.
+    $this->assertNotMenuLink($ogmenu_instance_ids[1], $custom_page->id());
+    // Check that a menu link has been created in the 2nd collection.
+    $this->assertMenuLink($ogmenu_instance_ids[2], $custom_page->id());
+
+    $child_link = MenuLinkContent::load($child_link->id());
+    // Check that the child custom page parent relation has been removed.
+    $this->assertTrue($child_link->get('parent')->isEmpty());
+  }
+
+  /**
+   * Asserts that a menu link exists.
+   *
+   * @param int $og_menu_instance_id
+   *   The OG menu instance ID.
+   * @param int $nid
+   *   The targeted custom page node ID.
+   * @param int|null $occurrences
+   *   (optional) Number of expected occurrences. If not passed, this test will
+   *   assume only one occurence.
+   *
+   * @return \Drupal\menu_link_content\MenuLinkContentInterface|null
+   *   The menu link content entity or NULL.
+   */
+  protected function assertMenuLink($og_menu_instance_id, $nid, $occurrences = 1) {
+    $properties = [
+      'bundle' => 'menu_link_content',
+      'menu_name' => "ogmenu-{$og_menu_instance_id}",
+      'link__uri' => "internal:/node/$nid",
+    ];
+    $storage = $this->container->get('entity_type.manager')
+      ->getStorage('menu_link_content');
+    $links = $storage->loadByProperties($properties);
+    $this->assertNotEmpty($links);
+    $this->assertCount($occurrences, $links);
+    return reset($links);
+  }
+
+  /**
+   * Asserts that a menu link doesn't exist.
+   *
+   * @param int $og_menu_instance_id
+   *   The OG menu instance ID.
+   * @param int $nid
+   *   The targeted custom page node ID.
+   */
+  protected function assertNotMenuLink($og_menu_instance_id, $nid) {
+    $properties = [
+      'bundle' => 'menu_link_content',
+      'menu_name' => "ogmenu-{$og_menu_instance_id}",
+      'link__uri' => "internal:/node/$nid",
+    ];
+    $storage = $this->container->get('entity_type.manager')
+      ->getStorage('menu_link_content');
+    $links = $storage->loadByProperties($properties);
+    $this->assertEmpty($links);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function tearDown() {
+    foreach ([1, 2] as $index) {
+      Rdf::load("http://example.com/$index")->delete();
+    }
```

Comment:
The `parent::tearDown()` should also be called here, it does a lot of important clean up work that preserves memory during the test runs.